src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +839,207 @@ public function in_default_dir( $src ) {
+	 *
+	 * @since 6.3.0
+	 * @see WP_Scripts::should_delay_inline_script()
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			foreach ( array( 'before', 'after' ) as $position ) {
+				if (
+					$this->get_data( $handle, $position ) &&
+					$this->should_delay_inline_script( $handle, $position )
+				) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Gets all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return string[] Script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
+			return $this->dependents_map[ $handle ];
+		}
+
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+
+		// Add the handles dependents to the map to ease future lookups.
+		$this->dependents_map[ $handle ] = $dependents;
+
+		return $dependents;
+	}
+
+	/**
+	 * Checks if the strategy passed is a valid delayed (non-blocking) strategy.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategies, otherwise false.
+	 */
+	private function is_delayed_strategy( $strategy ) {
+		return in_array(
+			$strategy,
+			$this->delayed_strategies,
+			true
+		);
+	}
+
+	/**
+	 * Checks if all of a script's dependents are delayed, which is required to maintain execution order.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string   $handle     The script handle.
+	 * @param bool     $async_only Whether to limit to async strategy only.
+	 * @param string[] $checked    Optional. An array of already checked script handles, used to avoid recursive loops.
+	 * @return bool True if all dependents are delayed, false otherwise.
+	 */
+	private function has_only_delayed_dependents( $handle, $async_only = false, $checked = array() ) {
+		// If this node was already checked, this script can be deferred and the branch ends.
+		if ( in_array( $handle, $checked, true ) ) {
+			return true;
+		}
+
+		$checked[]  = $handle;
+		$dependents = $this->get_dependents( $handle );
+
+		// If there are no dependents remaining to consider, the script can be deferred.
+		if ( empty( $dependents ) ) {
+			return true;
+		}
+
+		// Consider each dependent and check if it is delayed.
+		foreach ( $dependents as $dependent ) {
+			// If the dependent script has no src, ignore it for consideration.
+			if ( empty( $this->registered[ $dependent ]->src ) ) {
+				continue;
+			}
```

Comment:
@felixarntz This was introduced in 76e3ac8707a85cc85b91f7ef8e5af6725c28a2bd but what was the scenario that necessitated it? Curious how it was uncovered and I didn't see a test case. 

Moreover, it doesn't seem that `continue` is right here and that it should instead `return false`. This is because a script bundle (where `src` is `false`) is considered always blocking in `get_eligible_loading_strategy`.

Reply:
Related to https://github.com/10up/wordpress-develop/issues/76

Reply:
@westonruter Without this fix it was for example impossible to make jQuery in WP core deferred. Of course that's not something we specifically want to do (way too much risk), but we tested it, and something like that needs to be _possible_.

The problem was that `jquery` is an empty dependency that relies on `jquery-core` and `jquery-migrate`. I can set both of the latter to `defer`, but I can't do the same for `jquery` since it's prohibited by the current logic. So I was thinking that we need to ensure empty dependencies are ignored when determining whether a dependency of it is eligible for e.g. `defer`.

> Moreover, it doesn't seem that `continue` is right here and that it should instead `return false`. This is because a script bundle (where `src` is `false`) is considered always blocking in `get_eligible_loading_strategy`. 

That's the part I don't understand, and I'd currently disagree with. For example, `jquery` as a dependency has zero code, so we should be able to completely ignore it when determined the eligible loading strategy for its dependencies. A related question is: Why do we even prevent passing a strategy for empty dependencies? I probably missed a conversation somewhere, but what's the rationale?

Reply:
The issue is that even though such a dependency has zero code in it, it may still have before/after inline scripts. If such zero-code dependency gets a delayed strategy, any `after` inline script would break because there wouldn't be any script `load` event to listen for.

Also, what happens if even though the zero/bundle dependency is marked with a delayed strategy, it has dependencies which aren't? That's not currently accounted for here either.

Reply:
@felixarntz I believe I have a working solution here. See https://github.com/10up/wordpress-develop/pull/80

If you want to make jQuery deferred, what you can do is:

```php
foreach ( $wp_scripts->registered['jquery']->deps as $jquery_dep ) {
	$wp_scripts->registered[ $jquery_dep ]->add_data( 'strategy', 'defer' );
}
```

PR includes test case:

https://github.com/10up/wordpress-develop/blob/bbec078b275a64b582d9cc23a54b168dc7632562/tests/phpunit/tests/dependencies/scripts.php#L1051-L1066